src/services/firestoreServices.tsx

```diff
@@ -82,6 +115,43 @@ export const getDocuments = async <T,>(
   }
 };
 
+export const getUniqueCategories = async (
+  collectionName: string
+): Promise<CategoryWithImage[]> => {
+  try {
+    const collectionRef = collection(db, collectionName);
+    const querySnapshot = await getDocs(collectionRef);
+
+    const categoryImagesMap: Map<string, string[]> = new Map();
+
+    querySnapshot.forEach((doc) => {
+      const data = doc.data();
+      const categories = data["categories"] || [];
```

Comment:
I would suggest updating this part, when the data.categories or data.image is undefined. you'll ensure that it won't throw an error

const categories = data?.categories ?? [];
const imageUrl = data?.image ?? "";